scripts/getPVAliases.py

```diff
@@ -0,0 +1,508 @@
+        confirm = simple_prompt('Is '
+                                + Fore.LIGHTYELLOW_EX + f'{result} '
+                                + Style.RESET_ALL
+                                + 'correct? (y/N): ')
+        if confirm is True:
+            break
+    return result
+
+
+def flatten_list(input_list: list):
+    """
+    Flatten a list of lists and find its unique values.
+    Note: loses order due to set() call.
+    """
+    _result = [e for l in input_list for e in l]
+    return list(set(_result))
+
+
+def search_file(file:str,output:list=None,patt:str=None,prefix:str='',
+                color_wrap:Fore=None):
+    """
+    Searches file for regex match and appends result to list
+
+    Parameters
+    ----------
+    file : str
+        The file to read and search. Encoding must be utf-8
+    output : list, optional
+        A list to appead your results to. The default is None.
+    patt : str, optional
+        The regex pattern to search for. The default is None.
+    prefix : str, optional
+        A str prefix to add to each line. The default is ''.
+    color_wrap : Fore, optional
+        Color wrapping using Colorama.Fore. The default is None.
+
+    Returns
+    -------
+    list
+        A list of the search results with the prefix prepended.
+    """
+    if output is None:
+        output = []
+    color = ''
+    reset = ''
+    if color_wrap is not None:
+        color = color_wrap
+        reset = Style.RESET_ALL
+    if os.path.isfile(file) is False:
+        print(f'{file} does not exist')
+        return ''
+    with open(file,'r',encoding='utf-8') as _f:
+        for line in _f.readlines():
+            if re.search(patt,line):
+                output.append(re.sub(patt,color + r'\g<0>' + reset,line))
+        return prefix + prefix.join(output)
+
+
+def clean_ansi(text: str = None):
+    """
+    Removes ANSI escape sequences from a str, including fg/bg formatting.
+    """
+    ansi_escape = re.compile(r'\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])')
+    return ansi_escape.sub('',text)
+
+
+def fix_json(raw_data: str):
+    """
+    Fixes JSON format of find_ioc output.
+    """
+    # clean empty rows and white space
+    _temp = raw_data.replace(' ','').strip()
+    # capture and fix the keys not properly formatted to str
+    _temp = re.sub(r"(?<!'|\w)\w+(?!')(?=:\s?)",r"'\g<0>'",raw_data)
+    # capture boolean tokens and fix them for json format
+    _temp = re.sub("True","true",_temp)
+    _temp = re.sub("False","false",_temp)
+    # then capture and fix digits not formatted to str
+    _temp = re.sub(r"(?<=:)\d+",r"'\g<0>'",_temp)
+    # then properly convert to list of json obj
+    result = (_temp
+              .replace('\'','\"')
+              .replace('},','}')
+              .replace(' {','{')
+              .strip()
+              .split('\n'))
+    return result
+
+
+def find_ioc(hutch: str=None, patt: str=None):
+    """
+    A pythonic grep_ioc for gathering IOC details from the cfg file
```

Comment:
I very much wanted to avoid `iocmanager`, but was stubborn about the `ioc.cfg` outputs which were just so very nearly `json`.  Doing it this way let me use regex liberally, although it should absolutely have restraint haha.

Reply:
On the flip side, this code becomes much easier to maintain if we replicate the native parsing. I could imagine how all of the replace/strip/regex done here may run into some unknown edge case later down the line.

I personally would advocate for changing iocmanager's format to a real json structure but that feels out of scope here.

Reply:
I'm a huge fan of that as well, but I agree it's outside of the scope for this script sadly.

I know regex is a spice and it's easy to over season, but I quite like the option of regex searching the innards of the `ioc.cfg` dicts as it can make it much easier for catching bulk info in the MODs, e.g. `lm2k2.*qmini`. Maybe this is also my laziness speaking and it's not a good reason to keep it in, but wouldn't the `execfile` route require you to know/provide the full ioc name explicitly? 

This is probably safer, tbh, since you can't accidentally grab info you don't want. But on the other hand, naming conventions for MCS2 controllers seldom include MODs/TILE info that is relevant (also, sometimes TILEs are stretch across 2 IOCs):
![image](https://github.com/pcdshub/engineering_tools/assets/149725219/3d25b1c6-2e82-4dbe-bb83-c7a34a350be2)

So for fixing all the HRM aliases you'd probably want to catch both these IOCs.

What's the smallest hammer I can use here for the same endgoal?

Reply:
I think regex is probably a good way to tackle the `ioc.cfg` files, I think we're specifically proposing parsing the iocmanager config file with `execfile`.  Running `execfile` on the individual `ioc-*.cfg` doesn't itself make much sense.